### ISSUE TITLE
CLMM v2.0 desired features

### ISSUE THREAD

**Issue by combet** [LSSTDESC/CLMM#474]
This is an issue to discuss what functionalities we'd like to see in CLMM v2.0

**Comment by combet** [comment]
Top of my head at the moment:
- Stack analysis (#457, #460, #462)
- Einasto/Hernquist profiles for CCL backend (#461)
- 2 halo term (#459 )
- Include mass fitting in CLMM (#456)
- Handle shear calibration and metadetect outputs

**Comment by cavestruz** [comment]
Questions/Comments:
- What do folks feel the scope of the minimal fitting module should be?   I remember there being previous discussions on how much the fitting should rely on the user.  But, if this is framed as a wrapper function within which folks can swap in/out their favorite fitter with 1-2 defaults, I could see this being useful.  
- [convener hat on]  Can we define the CLMM v2.0 project, and can folks present this in the next CL General? :-)

**Comment by cavestruz** [comment]
Initial suggestions from CLMM meeting for fitter:
- Scipy optimizers (LGB....) specify box constraints instead of open ended paramaters, and set pseudo-priors (Angus)
- emcee (can do things in parallel; all we need to provide is likelihood) - need to be able to do things like ensure that chains converged, etc.  Note, can be tricky if we want automation - will require user to have more control.   (Michel)

**Comment by combet** [comment]
Other points raised during the CLMM discussion:
- Regarding MCMC, package like `ChainConsumer` is very useful to check properties of chains, convergence, etc. (Angus)
- Modeling: allow for triaxial/ellipsoidal density profiles (Calum)
- Covariance: we have covariance computed for a cluster ensemble, but might want to also include functionality for single cluster analysis (Constantin). (Also linked to [that paper](https://arxiv.org/pdf/1907.06611.pdf) from Heidi)
- Boost estimation:
  - Parametric implementation ongoing but how do we actually define the values of those parameters?
  - Calum: this is actually tricky and can be very dependent on photoz algorithm used
  - Michel: but can probably be infer from N(z) of the cluster versus background
  - Angus: only if you assume you have the true N(z), which we don't.
  - ...[Céline: not sure I've had all this conversation right, please edit accordingly @m-aguena, @calumhrmurray]
- Interfacing with `qp`
  - `qp` will be the way photoz information is stored and pass around in DESC 
  - ACTION ITEM: need to get in touch with `qp` folks and start adapting CLMM to handle photoz through `qp`

**Comment by combet** [comment]
Adding a new point to the list discussed above. 

- Shear calibration: we need to be able to handle `metadetect` outputs

**Comment by marina-ricci** [comment]
The previous comments cover what I had in mind. I add the following : 
- Would be nice to have methods to compute analytical covariances
- The fitting module, using MCMC, is a “must have” for v2.2. As a user story, I want to be able to infer a posterior on the mass of an ensemble of clusters (and individual clusters), given a data vector containing the profiles + covariance. 
- Having a method to predict reduced shear profiles, given a cosmology, survey specifications (for the source sample) and cluster properties (redshift + observable). This would be the first brick for a forward modeling approach.

**Comment by combet** [comment]
Regarding the fitting module "must have", @vitenti  just mentioned at the end of tag-up that the "connector" between FireCrown and CLMM to almost done, so we should be able to use FireCrown for this. Sandro, please correct me if I'm misunderstood.

**Comment by combet** [comment]
Oh and have CLMM pip-installable!
